### ISSUE TITLE
Multi-input gradients calculation on keras with K.function.

### ISSUE THREAD

**Issue by mabirck** [fchollet/keras#5353]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

`  
    #This model.input is a four head placeholder with size (3, 227, 227)
    input_imgs = model.input

    loss = K.max(layer_dict[layer_name].output[p[0]])

    # compute the gradient of the input picture wrt this loss
    grads = K.gradients(loss, input_imgs)[0]
    
    # this function returns the loss and grads given the input picture
    iterate = K.function(input_imgs, [grads])
    
`

I want to compute gradients for a siamese NN with fout head inputs, but when i apply the above K function I get a tensor of gradient with shape (1, 3, 227, 227), when what I want is (4, 3, 227, 227).

What can I do to get the correct gradients.
